web/pdf_find_controller.js

```diff
@@ -331,193 +462,140 @@ class PDFFindController {
     return true;
   }
 
-  /**
-   * Helper for multi-term search that fills the `matchesWithLength` array
-   * and handles cases where one search term includes another search term (for
-   * example, "tamed tame" or "this is"). It looks for intersecting terms in
-   * the `matches` and keeps elements with a longer match length.
-   */
-  _prepareMatches(matchesWithLength, matches, matchesLength) {
-    function isSubTerm(currentIndex) {
-      const currentElem = matchesWithLength[currentIndex];
-      const nextElem = matchesWithLength[currentIndex + 1];
-
-      // Check for cases like "TAMEd TAME".
-      if (
-        currentIndex < matchesWithLength.length - 1 &&
-        currentElem.match === nextElem.match
-      ) {
-        currentElem.skipped = true;
-        return true;
-      }
-
-      // Check for cases like "thIS IS".
-      for (let i = currentIndex - 1; i >= 0; i--) {
-        const prevElem = matchesWithLength[i];
-        if (prevElem.skipped) {
-          continue;
-        }
-        if (prevElem.match + prevElem.matchLength < currentElem.match) {
-          break;
-        }
-        if (
-          prevElem.match + prevElem.matchLength >=
-          currentElem.match + currentElem.matchLength
-        ) {
-          currentElem.skipped = true;
-          return true;
-        }
-      }
-      return false;
-    }
-
-    // Sort the array of `{ match: <match>, matchLength: <matchLength> }`
-    // objects on increasing index first and on the length otherwise.
-    matchesWithLength.sort(function (a, b) {
-      return a.match === b.match
-        ? a.matchLength - b.matchLength
-        : a.match - b.match;
-    });
-    for (let i = 0, len = matchesWithLength.length; i < len; i++) {
-      if (isSubTerm(i)) {
-        continue;
-      }
-      matches.push(matchesWithLength[i].match);
-      matchesLength.push(matchesWithLength[i].matchLength);
-    }
-  }
-
   /**
    * Determine if the search query constitutes a "whole word", by comparing the
    * first/last character type with the preceding/following character type.
    */
   _isEntireWord(content, startIdx, length) {
-    if (startIdx > 0) {
+    let match = content
+      .slice(0, startIdx)
```

Comment:
Here, and similar below, would it make sense to check that `match` isn't empty *before* running a then unnecessary regular expression?

Reply:
Do you mean checking that `startIdx` is not zero ?
If yes, I'd say it's useless since content is the page content and very likely startIdx is not zero, and I hope there is an optimization to immediately return an empty match in case the string is empty.
But of course if you think it's better to check, no problem, I'll do.